.github/workflows/build.yml

```diff
@@ -0,0 +1,44 @@
+name: Build OpenAPI docs
+on:
+  - push
+  # - pull_request
```

Comment:
I'm not sure. Isn't what we want more like:

```yaml
on:
  pull_request:
    types: [opened, synchronize, reopened]
  push:
    branches:
      - master
```

?

This would:
- Run exactly one instance of the workflow - with only the `docs-build` job - whenever a pull request is opened (against any target branch), updated (regardless of the number of commits in the update) or reopened after being closed.
- Run exactly once for a push against `master`, i.e., after a PR against `master` has been merged; in this case, due to `if: github.ref == 'refs/heads/master', also the `deploy` job would be run.

I think this would be less broad/more specific.

Alternatively, perhaps remove the `# - pull_request` line.

Reply:
I think we want the `pull_request` line since then we can have documentation for `master`, `development` and other branches to see issues in the documentation. Thoughts?

Reply:
I agree. But in the current workflow it's commented out in favor of every `push` (which is very broad) - in which case we really don't need `pull_request` at all.

My suggestion would run the workflow for every PR (opened, updated or reopened), but not for branches for which no PR exists. PLUS it would run _after_ merging changes into `master`. Would this be sufficient in your opinion, @vsmalladi?

Reply:
Ya i agree. Let me change that